File: objc/Modulino.m
#import <Foundation/Foundation.h>
#import "Modulino.h"

@implementation Modulino

+ (NSInteger) meaningOfLife {
    return 42;
}

@end

int __attribute__((weak)) main() {
    NSAutoreleasePool *pool = [[NSAutoreleasePool alloc] init];

    NSLog(
        @"%@",
        [NSString
            stringWithFormat:
            @"Main: The meaning of life is %ld",
            [Modulino meaningOfLife]
        ]
    );

    [pool drain];

    return 0;
}

File: swift/Modulino.swift
import Foundation

public class Modulino {
  public var meaningOfLife = 42

  public init() {}

  public class func main() {
    var meaning = Modulino().meaningOfLife

    println("Main: The meaning of life is \(meaning)")
  }
}

#if MODULINO
@objc class ModulinoAutoload {
  @objc class func load() {
    Modulino.main()
  }
}
#endif
